mopidy_musicbox_webclient/static/js/controls.js

```diff
@@ -588,22 +607,65 @@ function deleteFavourite(index) {
 function showFavourites() {
     $('#streamuristable').empty();
     getFavourites().then(function(favourites) {
-        if (favourites && favourites.tracks) {
-            tracks = favourites.tracks;
-            var tmp = '';
+        if (!favourites) {
+            return;
+        }
+        var tmp = '';
+        
+        $.cookie.json = true;
+        if ($.cookie('streamUris')) {
+            tmp = '<button class="btn" style="padding: 5px; width: 100%" type="button" onclick="return upgradeStreamUrisToFavourites();">Convert StreamUris</button>'
+        }
+        if (favourites.tracks) {
             var child = '';
-            for (var i = 0; i < tracks.length; i++) {
+            for (var i = 0; i < favourites.tracks.length; i++) {
                 child = '<li><span class="ui-icon ui-icon-delete ui-icon-shadow" style="float:right; margin: .5em; margin-top: .8em;"><a href="#" onclick="return deleteFavourite(\'' + i + '\');">&nbsp;</a></span>' +
                     '<i class="fa fa-rss" style="float: left; padding: .5em; padding-top: 1em;"></i>' +
-                    ' <a style="margin-left: 20px" href="#" onclick="return playStreamUri(\'' + tracks[i].uri + '\');">';
-                child += '<h1>' + tracks[i].name + '</h1></a></li>';
+                    ' <a style="margin-left: 20px" href="#" onclick="return playStreamUri(\'' + favourites.tracks[i].uri + '\');">';
+                child += '<h1>' + favourites.tracks[i].name + '</h1></a></li>';
                 tmp += child;
             }
-            $('#streamuristable').html(tmp);
         }
+        $('#streamuristable').html(tmp);
     });    
 }
 
+# TODO: Remove this upgrade path in next major release.
```

Comment:
This produces a syntax error. Should probably be:
```javascript
/* TODO: Remove this upgrade path in next major release. */
```
Apart from that I think we should merge.